### PR TITLE
Handle WS CLOSE properly

### LINKED ISSUE #1787: Writing to an already closed websocket connection throws errors and increases endpoint's response time
Here is how to reproduce:

1. Run the latest bigchaindb server locally
2. Install and download bigchaindb/benchmarks
3. Run a script that initiates a WS connection with BigchainDB and e.g. counts the messages receives

I've prepared a script already:

```javascript
(function replication() {
    const conns = []
    let messages = 0

    function onOpen() {
        console.log('open')
    }

    function onError(error) {
        if (error.code === 'EAI_AGAIN') {
            console.log('Connection problem with BDB')
        }
    }

    function onClose() {
        //console.log('close')
    }

    function onMessage(data) {
        messages += 1
        console.log(messages)
    }

    const listeners = {
        open: onOpen,
        error: onError,
        close: onClose,
        message: onMessage
    }

    function initSockets() {
        const ws = new WebSocket('ws://localhost:9985/api/v1/streams/valid_transactions')
        Object
            .keys(listeners)
            .forEach(key => ws.addListener(key, listeners[key]))

        conns.push(ws)
    }
    initSockets()
})()
```

4. Send a transaction using the benchmark tool to see if the script works

```
bigchaindb-benchmark -p http://localhost:9984 send -r 1
```

5. If it does. `1` should appear from the console

6. Now increase the number of transactions by several magnitudes such that your script will start to have problems receiving them

```
bigchaindb-benchmark -p http://localhost:9984 send -r 10000000
```

7. Wait for a few thousand messages to be sent, then quickly terminate the script with Ctrl-C

8. BigchainDB should now log this message many times:

```
(asyncio) socket.send() raised exception
```

Likely because the connection bigchaindb just wanted to write a transaction to just got closed by the script.

If you now try to reconnect to BigchainDB, its response time will be extremely high, to the point of connections not getting accepted anymore.

@vrde Happy to fix this myself if you tell me where.

### PR DESCRIPTION
**Pull request by vrde**
TL;DR: [Writing to an already closed websocket connection throws errors and increases endpoint's response time](https://github.com/bigchaindb/bigchaindb/issues/1787).

Fix #1787

### CODE REVIEW

**Comment by ttmc** [comment]
@vrde Could you add a description to this pull request so we understand the reason for it?

**Comment by ttmc** [comment]
Did you test this against what @TimDaub described in the original issue #1787 ?

**Comment by vrde** [comment]
> Did you test this against what @TimDaub described in the original issue #1787 ?

Yes, but it's not in a unit test.

**Comment by ttmc** [comment]
Okay, that's reassuring.

**Comment by ttmc** [comment]
@vrde wrote in an email:

> You can merge it even if coverage complains.

because unit tests were not written to test all the code paths in this PR. Normally we wouldn't do that, but I think @vrde is considering replacing all this ascyncio stuff with something else.
